File: LeetCode Problems/101 to 200/144. Binary Tree Preorder Traversal/144. Binary Tree Preorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private List<Integer> arr = new ArrayList<>();
    private void PreorderTraversal(TreeNode root) {
        if (root == null) return;
        arr.add(root.val);
        PreorderTraversal(root.left);
        PreorderTraversal(root.right);
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        PreorderTraversal(root);
        return arr;
    }
}

File: LeetCode Problems/101 to 200/144. Binary Tree Preorder Traversal/144. Binary Tree Preorder Traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    var arr = ArrayList<Int>();
    private fun PreorderTraversal(root: TreeNode?) {
        if(root == null)
            return;
        arr.add(root.`val`);
        PreorderTraversal(root.left);
        PreorderTraversal(root.right);
    }
    fun preorderTraversal(root: TreeNode?): List<Int> {
        PreorderTraversal(root);
        return arr;
    }
}